internal/factsengine/entities/fact_value.go

```diff
@@ -99,3 +109,44 @@ func ParseStringToFactValue(str string) FactValue {
 
 	return &FactValueString{Value: str}
 }
+
+// GetValue returns a value using a dot access key format from a FactValue.
+// Examples:
+// foo.bar.buz access the {"foo": {"bar": {"baz": "value"}}}
+// foo.0.buz access the {"foo": [{"buz": "value"}]}
+func GetValue(fact FactValue, values string) (FactValue, *FactGatheringError) {
```

Comment:
wouldn't be possible to expose this as method of `FactValueMap` and `FactValueList` so that in the corosync conf gahterer we just can call the method without changing the type to the interface in the `corosyncConfToMap` function?

Reply:
Agree with @fabriziosestito this should be a method on the structs

Reply:
mmm, let me check the option. I guess it should be totally doable.
PD: Fabriozio asking to move from "pure" functions to OOP kind of functions. Not sure if this was not written by copilot... XD

Reply:
The gopher daemon corrupts everyone, a "rustichella bite" after another 🤣

Reply:
Done!
I have attached the function only to `FactValueMap` even though it could be useful for lists as well, but we don't have this case yet, so we can think of adding it once we need it

Reply:
I can argue this is stil a pure function!